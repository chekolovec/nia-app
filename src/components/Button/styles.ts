import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    padding: 20,
    borderWidth: 1,
    borderRadius: 20,
    backgroundColor: "#007BA7",
    borderColor: "#007BA7",
  },
  text: {
    fontSize: 20,
    color: "#FFF",
  },
  pressed: {
    opacity: 0.7,
  },
});
