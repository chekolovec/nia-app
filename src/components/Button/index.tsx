import React from "react";
import { Pressable, Text } from "react-native";

import styles from "./styles";

interface IProps {
  title: string;
  onPress: () => void;
}

export const Button = ({ title, onPress }: IProps) => {
  return (
    <Pressable
      testID="pressHandler"
      onPress={onPress}
      style={({ pressed }) => [styles.container, pressed && styles.pressed]}
    >
      <Text testID="titleText" style={styles.text}>
        {title}
      </Text>
    </Pressable>
  );
};
