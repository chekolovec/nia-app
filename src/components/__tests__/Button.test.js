import * as React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";

import { Button } from "../Button";

describe("test", () => {
  it(`snapshots match`, () => {
    const tree = renderer.create(<Button title={"Test title"} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

it("given a title Button component should render it", () => {
  const { getByTestId } = render(
    <Button title="Test title" onPress={() => {}} />
  );
  const title = getByTestId("titleText");

  expect(title).toBeDefined();
  expect(title.props.children).toBe("Test title");
});

it("when Button pushed onPress should be called", () => {
  const mockFunc = jest.fn();
  const { getByTestId } = render(
    <Button title="Test title" onPress={mockFunc} />
  );

  fireEvent.press(getByTestId("pressHandler"));

  expect(mockFunc).toHaveBeenCalled();
});
