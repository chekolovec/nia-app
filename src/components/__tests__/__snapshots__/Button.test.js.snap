// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshots match 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#007BA7",
        "borderColor": "#007BA7",
        "borderRadius": 20,
        "borderWidth": 1,
        "padding": 20,
      },
      false,
    ]
  }
  testID="pressHandler"
>
  <Text
    style={
      Object {
        "color": "#FFF",
        "fontSize": 20,
      }
    }
    testID="titleText"
  >
    Test title
  </Text>
</View>
`;
